### PR TITLE
[google_maps_flutter] Clear duplicated dependencies 

### PR DESCRIPTION
**Pull request by kodal**
Remove duplicated dependencies according by [dart-lang/pub#2101]

No version change: No effect on plugin clients.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[dart-lang/pub#2101]: https://github.com/dart-lang/pub/issues/2101

### CODE REVIEW

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for the submission!

This needs a `CHANGELOG.md` update using `## NEXT` as the version, but otherwise looks good.

**Comment by kodal** [comment]
Thanks for review!

Added description to `CHANGELOG.md`.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.1
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:1]: This should not be a version bump as it does not affect customers of the package. It should use `NEXT`, per my earlier comment.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!

**Comment by stuartmorgan** [comment]
@cyanglaz Could you do the secondary review here?

**Comment by stuartmorgan** [comment]
Tagging @bparrishMines for secondary review of this trivial cleanup.

**Comment by stuartmorgan** [comment]
@cyanglaz @bparrishMines Ping for one of you to sign off on this per two-review policy?

**Comment by stuartmorgan** [comment]
@kodal Some of the tests never started for some reason; could you merge in the latest `master` to kick off a new round of tests?

**Review by cyanglaz — APPROVED** [review]
LGTM

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
@kodal Are you planning on merging in master so we can land this? I'm not able to do so from the GitHub UI any more for some reason.

**Pull request merged**